lib/vdc/BOVCollection.cpp

```diff
@@ -473,28 +514,26 @@ template<class T> int BOVCollection::ReadRegion(std::string varname, size_t ts,
         return -1;
     }
 
-    size_t numValues = _gridSize[0] * _gridSize[1] * _gridSize[2];
-
-    int  n = 1;
-    bool systemLittleEndian = *(char *)&n == 1 ? true : false;
-    bool dataLittleEndian = _dataEndian == "LITTLE" ? true : false;
-    bool needSwap = systemLittleEndian != dataLittleEndian ? true : false;
-
     // Read a "pencil" of data along the X axis, one row at a time
     size_t count = max[0] - min[0] + 1;
     // Note: allocate buffer once and reuse for many times, so repeated allocation is avoided.
     std::vector<unsigned char> vReadBuffer(count * formatSize);
     unsigned char *            readBuffer = vReadBuffer.data();
+
     for (size_t k = min[2]; k <= max[2]; k++) {
         size_t zOffset = _gridSize[0] * _gridSize[1] * k;
         for (size_t j = min[1]; j <= max[1]; j++) {
             size_t xOffset = min[0];
             size_t yOffset = _gridSize[0] * j;
             size_t offset = formatSize * (xOffset + yOffset + zOffset) + _byteOffset;
 
-            fseek(fp, offset, SEEK_SET);
-            size_t rc = fread(readBuffer, formatSize, count, fp);
+            size_t rc = fseek(fp, offset, SEEK_SET);
```

Comment:
Scott, the function fseek returns a type of signed int. Please, don't do a signed -> unsigned conversion, please.

Reply:
after you fix this, please check other function return types, and make sure you don't make unintended conversions between signed and unsigned integers.

Reply:
You're right.  Fixed.  Pushing soon.

Reply:
Scott, 5 lines down, fread() will return a size_t type. Can you declare a separate variable with the proper type instead of re-using the same variable?

Reply:
Thanks, fixed.